exasol/bucketfs/_path.py

```diff
@@ -376,3 +385,113 @@ def __truediv__(self, other):
 
     def __str__(self):
         return str(self._path)
+
+
+def _create_onprem_bucket(url: str,
+                          username: str,
+                          password: str,
+                          bucket_name: str = 'default',
+                          verify: bool | str = True
+                          ) -> BucketLike:
+    """
+    Creates an on-prem bucket.
+    """
+    credentials = {bucket_name: {'username': username, 'password': password}}
+    service = Service(url, credentials, verify)
+    buckets = service.buckets
+    if bucket_name not in buckets:
+        raise BucketFsError(f'Bucket {bucket_name} does not exist.')
+    return buckets[bucket_name]
+
+
+def _create_saas_bucket(account_id: str,
+                        database_id: str,
+                        pat: str,
+                        url: str = 'https://cloud.exasol.com'
+                        ) -> BucketLike:
+    """
+    Creates a SaaS bucket.
+    """
+    return SaaSBucket(url=url, account_id=account_id, database_id=database_id, pat=pat)
+
+
+def _create_mounted_bucket(service_name: str = 'bfsdefault',
+                           bucket_name: str = 'default',
+                           ) -> BucketLike:
+    """
+    Creates a bucket mounted to a UDF.
+    """
+    bucket = MountedBucket(service_name, bucket_name)
+    if not bucket.root.exists():
+        raise BucketFsError(f'Service {service_name} or bucket {bucket_name} do not exist.')
+    return bucket
+
+
+def build_path(**kwargs) -> PathLike:
+    """
```

Comment:
mock path is missing

Reply:
What is the "mock path"?

Reply:
you might remember our discussion about how we configure UDFs that need to access files in the BucketFS, we do this via a connection object. If we now test a UDF implementation locally via the udf-mock-python, this is still the case, but in that case we usually supply in the connection object a temporary directory in the local file system.

Reply:
Would the answer be pushing this to the user? The user would provide the root directory, which would normally be `buckets/service_name/bucket_name`, but can instead be a local directory. Or we can give a choice of providing either an explicit root directory or a combination of the `service_name` and the `bucket_name`. 
This would imply an answer to the question of what should be seen as the base directory - the names of the bucket and the service would be excluded.